src/app/services/camera.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, OnDestroy, signal } from '@angular/core';
+import { gamestate, setCameraPosition } from '../helpers';
+
+export interface Point {
+  x: number;
+  y: number;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CameraService implements OnDestroy {
+  public isDragging = signal<boolean>(false);
+  public lastPanPosition = signal<Point | null>(null);
+
+  #boundOnMouseDown: (event: MouseEvent) => void;
+  #boundOnMouseMove: (event: MouseEvent) => void;
+  #boundOnMouseUp: () => void;
+  #panSensitivity = 2;
+
+  constructor() {
+    this.#boundOnMouseDown = this.onMouseDown.bind(this);
+    this.#boundOnMouseMove = this.onMouseMove.bind(this);
+    this.#boundOnMouseUp = this.onMouseUp.bind(this);
+  }
+
+  public init(): void {
+    this.setupMouseListeners();
+  }
+
+  ngOnDestroy(): void {
+    if (typeof window !== 'undefined') {
+      window.removeEventListener('mousedown', this.#boundOnMouseDown);
+      window.removeEventListener('mousemove', this.#boundOnMouseMove);
+      window.removeEventListener('mouseup', this.#boundOnMouseUp);
+    }
+  }
+
+  private setupMouseListeners(): void {
+    if (typeof window !== 'undefined') {
```

Comment:
I would prefer to see this as a guard, eg, `if typeof window === undefined return` rather than putting the contents of the function in an if